genesyscloud/telephony_providers_edges_edge_group/genesyscloud_telephony_providers_edges_edge_group_proxy.go

```diff
@@ -0,0 +1,118 @@
+package telephony_providers_edges_edge_group
+
+import (
+	"context"
+
+	"github.com/mypurecloud/platform-client-sdk-go/v121/platformclientv2"
+)
+
+var internalProxy *edgeGroupProxy
+
+type getEdgeGroupFunc func(ctx context.Context, p *edgeGroupProxy, edgeGroupId string) (*platformclientv2.Edgegroup, *platformclientv2.APIResponse, error)
+type deleteEdgeGroupFunc func(ctx context.Context, p *edgeGroupProxy, edgeGroupId string) (*platformclientv2.APIResponse, error)
+type putEdgeGroupFunc func(ctx context.Context, p *edgeGroupProxy, edgeGroupId string, body platformclientv2.Edgegroup) (*platformclientv2.Edgegroup, *platformclientv2.APIResponse, error)
+type postEdgeGroupFunc func(ctx context.Context, p *edgeGroupProxy, body platformclientv2.Edgegroup) (*platformclientv2.Edgegroup, *platformclientv2.APIResponse, error)
+type getAllEdgeGroupsFunc func(ctx context.Context, p *edgeGroupProxy) (*[]platformclientv2.Edgegroup, error)
+
+type edgeGroupProxy struct {
+	clientConfig *platformclientv2.Configuration
+	edgesApi     *platformclientv2.TelephonyProvidersEdgeApi
+
+	getEdgeGroupAttr     getEdgeGroupFunc
+	deleteEdgeGroupAttr  deleteEdgeGroupFunc
+	putEdgeGroupAttr     putEdgeGroupFunc
+	postEdgeGroupAttr    postEdgeGroupFunc
+	getAllEdgeGroupsAttr getAllEdgeGroupsFunc
+}
+
+func newEdgeGroupProxy(clientConfig *platformclientv2.Configuration) *edgeGroupProxy {
+	edgesApi := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(clientConfig)
+
+	return &edgeGroupProxy{
+		clientConfig: clientConfig,
+		edgesApi:     edgesApi,
+
+		getEdgeGroupAttr:     getEdgeGroupFn,
+		deleteEdgeGroupAttr:  deleteEdgeGroupFn,
+		putEdgeGroupAttr:     putEdgeGroupFn,
+		postEdgeGroupAttr:    postEdgeGroupFn,
+		getAllEdgeGroupsAttr: getAllEdgeGroupsFn,
+	}
+}
+
+func getEdgeGroupProxy(clientConfig *platformclientv2.Configuration) *edgeGroupProxy {
+	if internalProxy == nil {
+		internalProxy = newEdgeGroupProxy(clientConfig)
+	}
+	return internalProxy
+}
+
+func (p *edgeGroupProxy) getEdgeGroup(ctx context.Context, edgeGroupId string) (*platformclientv2.Edgegroup, *platformclientv2.APIResponse, error) {
+	return p.getEdgeGroupAttr(ctx, p, edgeGroupId)
+}
+
+func (p *edgeGroupProxy) deleteEdgeGroup(ctx context.Context, edgeGroupId string) (*platformclientv2.APIResponse, error) {
+	return p.deleteEdgeGroupAttr(ctx, p, edgeGroupId)
+}
+
+func (p *edgeGroupProxy) putEdgeGroup(ctx context.Context, edgeGroupId string, body platformclientv2.Edgegroup) (*platformclientv2.Edgegroup, *platformclientv2.APIResponse, error) {
+	return p.putEdgeGroupAttr(ctx, p, edgeGroupId, body)
+}
+
+func (p *edgeGroupProxy) postEdgeGroup(ctx context.Context, body platformclientv2.Edgegroup) (*platformclientv2.Edgegroup, *platformclientv2.APIResponse, error) {
```

Comment:
we generally call the post functions `createEdgeGroup` instead, same with `putEdgeGroup` should be update

Reply:
addressed